### PR TITLE
ENH: Add StringMethods.isnumeric and isdecimal

### PR DESCRIPTION
**Pull request by sinhrks**
Derived from #9111.

### CODE REVIEW

File: pandas/tests/test_strings.py [hunk 1]
```diff
@@ -659,6 +661,31 @@ def test_ismethods(self):
         self.assertEquals(str_s.str.isupper().tolist(), [v.isupper() for v in values])
         self.assertEquals(str_s.str.istitle().tolist(), [v.istitle() for v in values])
 
+    def test_isnumeric(self):
+        # 0x00bc: ¼ VULGAR FRACTION ONE QUARTER
+        # 0x2605: ★ not number
+        # 0x1378: ፸ ETHIOPIC NUMBER SEVENTY
+        # 0xFF13: ３ Em 3
+        values = ['A', '3', unichr(0x00bc), unichr(0x2605),
```
Comment (shoyer) [inline pandas/tests/test_strings.py]: ugh... I guess we still support Python 3.2 :(

Comment (sinhrks) [inline pandas/tests/test_strings.py]: Isn't 3.2 tested in below? I've expected ``compat.unichr`` can work for 3.2 also.

https://travis-ci.org/pydata/pandas/jobs/49806158

Comment (shoyer) [inline pandas/tests/test_strings.py]: I know, it works!

I was going to suggest just using `u'¼'` but the `u` unicode literal is not supported on Python 3.2.

Comment (sinhrks) [inline pandas/tests/test_strings.py]: I see. I also tried to use `u` at first and failed... Will wait a conclusion of #9118.

Comment (jreback) [inline pandas/tests/test_strings.py]: u(...) is the way to handle 3.2 compat

Comment (sinhrks) [inline pandas/tests/test_strings.py]: I think we can't use `u( )` here because escaped unicode is not handled as numeric or decimal.

**Comment by jreback** [comment]
thanks!

**Pull request merged**